### ISSUE TITLE
Graphics issues in downloaded maps

### ISSUE THREAD

**Issue by qina83** [mapbox/mapbox-gl-js#10681]
**mapbox-gl-js version**:
2.0.1

**browser**:
cross-platform-app in cordova.ionic-angular 

### Steps to Trigger Behavior

 1.  download multiple map for offline use
 2.  put offline the device
 3.  check offline map

Using this configuration source it presents graphics issues like the screen below
 
“offline-sources”: {
    "type": “raster”,
    "tiles": [
        “${LOCAL_PATH}/${REGION_1}/{z}/{x}/{y}.jpg”,
        “${LOCAL_PATH}/${REGION_2}/{z}/{x}/{y}.jpg”,
    ],
}

If instead I provide only one path to sources.tiles like this (example below) the tiles are displayed correctly without map issues.

“offline-sources”: {
    "type": “raster”,
    "tiles": [
        “${LOCAL_PATH}/${REGION}/{z}/{x}/{y}.jpg”,
    ],
}

### Expected Behavior
All downloaded maps area must be showed correctly

### Actual Behavior
<img width="242" alt="Screenshot 2021-05-13 at 17 47 30" src="https://user-images.githubusercontent.com/15413007/118151574-0cade280-b414-11eb-8008-b842a4c13da0.png">

**Comment by karimnaaji** [comment]
@qina83 It looks like the regions displayed are rendered with tiles at the wrong zoom level. If you could provide the content of these folders (just the name of the tiles) that would help a bit in assessing the issue and finding why wrong tiles are prioritized over. One thing to confirm is whether you are confident that the tiles provided in `tiles: []` are covering mutually exclusive regions?

**Comment by qina83** [comment]
Hi @karimnaaji, thanks for the reply.

this is the structure of tiles:
- the root folder is the region (1620830157741  and 1620830207752)
- the content of levels is shown in the images below
<img width="258" alt="Screenshot 2021-05-14 at 10 01 55" src="https://user-images.githubusercontent.com/15413007/118243505-cea9d080-b49e-11eb-8a70-8197cebc70a1.png">
![image](https://user-images.githubusercontent.com/15413007/118243520-d1a4c100-b49e-11eb-8e8b-01f69a0551a3.png)


The regions downloaded are mutually exclusive but for some level of zoom (i.e. 0) the are obviously overlapped like the screen below.
![image](https://user-images.githubusercontent.com/15413007/118243482-c782c280-b49e-11eb-8a10-547dbbbf6f45.png)

**Comment by qina83** [comment]
Hi, are there any news or suggestion about why this happens?
Thx

**Comment by karimnaaji** [comment]
Nothing jumps as out of the ordinary from your structure to me, the only thing I can't confirm is whether there's some overlap in the regions that would lead to incorrect prioritization.

Would you have a small region and example you could share where this is reproduced? A zip file or minimal demo page to run would be great.

**Comment by qina83** [comment]
Hi,
You would find the two screenshot of downloaded areas in the directory “downloaded-zones”, and the relative jpg files in “data/offline-regions”.
You can run the example in zip file using
- npm i
- npm start

Here the bug screenshots
![image](https://user-images.githubusercontent.com/15413007/119159477-8b7bce80-ba57-11eb-90ab-385781813a67.png)
![image (1)](https://user-images.githubusercontent.com/15413007/119159493-8fa7ec00-ba57-11eb-8eff-3f1eac8a8593.png)

If you want to make it work without seeing the issue, just remove one id (ex. “1621606307090”) from regionsIds

https://drive.google.com/file/d/1TsKeskoe95e9h2LZ0bKoutNlhl537n0a/view?usp=sharing

Thx

**Comment by karimnaaji** [comment]
Thanks for providing the data it was helpful to debug the behavior, so your data sources given within `tiles:[]` are incorrect. As mentioned [here](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#vector-tiles) it must follow the tileJSON spec which mentions the following:
```
// REQUIRED. An array of tile endpoints. {z}, {x} and {y} are replaced with
// the corresponding integers. If multiple endpoints are specified, clients
// may use any combination of endpoints. All endpoints MUST return the same
// content for the same URL. The array MUST contain at least one endpoint.
"tiles": [
  "http://localhost:8888/admin/1.0.0/world-light,broadband/{z}/{x}/{y}.png"
],
```

Refer https://github.com/mapbox/tilejson-spec/blob/master/2.0.1/README.md.

The data within your folders in `offline-regions` has to return the same content given a URL, in this case there are things available in `1621606172331` but not `1621606172331` and vice versa. I was incorrect when mentioning that they had to be mutually exclusive and confirmed how their structure had to be from the docs. My suggestion is to reorganize these data folders to be the same.

The behavior is correct from the code in the library, so I'll go a head and close this ticket.
